pages/api/v1/calendar/event.ts

```diff
@@ -3,13 +3,25 @@ import { calendar } from "@/util";
 import type { Data } from "@/util/types";
 
 export default async function handler(req: NextApiRequest, res: NextApiResponse<Data>) {
-  // Validate POST request type
-  if (req.method !== "POST") return res.status(400).json({ error: "Invalid request method" });
-
   // Validate body provided
   if (!req.body) return res.status(400).json({ error: "No body provided" });
-  const { name, start, end, description, location, eventLink } = req.body;
+  
+  // Validate request type
+  switch (req.method) {
+    case "POST":
+      return post_handler(req, res);
+    case "PATCH":
+      return patch_handler(req, res);
+    case "DELETE":
+      return delete_handler(req, res);
+    default:
+      return res.status(405).json({ error: "Invalid request method" });
```

Comment:
Just curious: Why 405 instead of 400?

Reply:
405 is method not allowed, so its a more specific error (it'll tell them the exact issue - that we don't support whatever method they used, instead of a broad 400 error)